### DISCUSSION TITLE
polyRoundExtrude with internal holes

### DISCUSSION THREAD

**Discussion by echo-bravo-yahoo** [Irev-Dev/Round-Anything#25] [Q&A] [answered]
Is there a way to use polyRoundExtrude on a polygon set with internal holes (e.g., a doughnut)?

If not, is there a way to achieve a similar effect of top+bottom chamfers (rounding X/Y/Z) and rounded edges (X/Y)? `polyRound` smooths in (X/Y), but then extruding it doesn't have a convient way to chamfer, and `polyRoundExtrude` can't take in the output of `polyRound`...

I may source-dive and cobble something together later, if I have the time...

**Comment by Irev-Dev** [comment, marked as answer]
Hey @Trial-In-Error,
Yeah it is supported in an indirect way, but I should probably document the intention better though.

`polyRoundExtrude` supports negative fillets, you have to then put on your normal OpenSCAD thinking hat and use difference with another [polyRound like so](https://cadhub.xyz/dev-ide/openScad#encoded_script_v2=eJztPWuT28aRn7W/YqJKZIIEdwFQsmMqsEqRo7PqFNu10Z2TsFg5kACXsEEABsBd0rH++/VjZjADgNyV7NTVVUUlrYBBT09PT3dPd89jqyhO02+LNG/qcLHwPddzfW/pLoJP3QAeL314/j09YqnnfuZ6lzN4eubO4Am/Tp9ineXy+UXV4vo6uYma9DZBnL6LaAHSf+biX6qFpfwMFXMJ/TrNsqSpXxfVV0WWhFP/8tnzi+TQVPs4+SpJb7ZNGDy/iNPNJqmSfJ2MnH9eCPhTFtnxutjn8Z8YdmRQ4lr1geZn9G+Th793nlPtporyOouaZLTA7k+9S897tnTOIVXds5FPoKbvDnfmVLGk4/0F/Lnq/4Ey8SqKv9qvRFwkdf5JI+p9WRZVI3b7rEmnmzRLRBlVTS2OCXwsxDa6TURTiO+jnWi2iaiSuhHFhp6zdFVF1VGskqy4G2yP/0Crbxl0LirkwTTKj802zW/w038nVZ0W+Vz4lx6+v9w326KaizdVcvtlcks0F3lTpat9U1T1XLx7+8oPZlxeHivk1VwEXuBxQ+skr5O5+PObd9D0Zp+vG8Auojj++7ooqnhU8kDGaV1m0TrZJXnjhItNUY3ScOHNsySXIM7UXzoLfl6ky8uD2z4fXWEiQLHTTWkJmH1JLdaqSfwQklSA/CLDwwXILUj7ufZBqEYGDWOsd3lwJogMSLI/H/nzUX4+dj7/xJ9/kp9/urDoLjabOmm+BUG9KXKWTEW5/BZ6TohM/q6ofhA1ikAd7RIR1SQODEOS5YrksE7KRkC36NsmWjfwEDUizesmiWKUoUgEsai3UZkg0jeNuAO8NTC2StZNdhRAU8nUCKZDglVJs69ybpQIyPe7VVIhSoYT3+9JSokeKChEUbkiuouOYlMVLMgFCE6aRxkTcImouXIIdNVJlSY1Ijy4R1m+WBx894jW5hC4x2DpXl5eLrGa5I2IdiDbDSqLbHkFUqIUVdJYi5sCKLqLQCyYMuoEkuBCO3XahS32DbbBXf4w0sAwjGD80/oVjNerV9+FdXqTj5g2ZywLlayNpz4KS1aGhgBeOBeGYJYoj2gdIrA52esiA60foXxKIrK0br6ronKUAq6sdJZu+y01X1rAiQJcumAP1j+ETJ27K+Ik1IQ7IKWtmIJZAN4kbwHJCDE5IhQLIclE4rlwKvz51J97QDIWAAUC0aCBKUsQrv8x8PyPkAMRVRXICPAWPjXJAUYoBntSkxhHefsZxDmpW5LSHOFfY+GIPrU0bUJZYJK9cYTVpV30Q/JqX90m8bdge7/ZoA5+lcRVkVtzTwX23d2l+XWIUwOrIrANBKkGwSir9BaMjmjtEKrbm092Ii8a0McSlApMrrba+xpFDDoOD1rnSNaMSZzErU5Y3ATIc2WoYhUS7J6kcY30c5t3MCnBpCDKfSOIjkQkeaxmDZrk0OIjik2u9QY+1/tVnN6m+JEaoL6KdANYElVdaniam3QiB7KkRpMAwwS6njafQEn6QwIjDV1cFzhCNzDAQAcMyw4VaRtVpUjim8TFtmjuBvw7iRl7RbRD71X1mMBrsYZGVlTcVDDtxtDjZkvUulCj2N9s4fM62jPjMjAB6yMIQFRDv4A0rA9TL2DwXAGzmsIXodgdRb0DBZM24NJSfiV+CzSLWxIQKRriW11C8oZ0JG+T/AboIktbw9DDjB0dwU6mPCY0YtpMSM03mOqgytJ72LUCBhTNK6TdWl+/gW5AF0JlYro4QdjYFlX0Gq3q6xB+yNdNFZH8frPZvC1Df+pfgdQjWJYpz5IUS4IhUQgyB7LIAZP9wT9pA4RIXGiioeFx/WPVjOofR9iqM4UHhWhMJc4UgfDRlUiAVd/ImZFaH5omTS3tNOooPNoDtLB1OSuBrW5wbQR/uYL5ACwF+KvgPBife2QOW+PT4Ok933s224C2qmopeFuso1YG+vQ7dq03OfAN5uNr0r1wZOMZd4XLCUOesdQfsoWJUt0c3AIwihUoYI4moNonZMT1FFt16t5tk5zgK7QletZO8zXqLSgP00Wmdt3sgYKjiBP1kezKLUwV09vaxttF8B1Yim8lEdeIjVwVMFHQrVXBBgPsQFSl0Qp8cewHEb47ojlDl9lgWgd3uIsOLIownNOOHLpknyymt/QP1J8M19fV26dFX5bAMR2QMPSauwP9oteLeY8uJWvLVpUQIwQ972D2e4s2LdTaNRrUM5g5mWD+2Y0GTFSubRAuZLtdkxtuwMXHnrxEqzzS1smRtuot21O7nOwvm2X76wL8FAIoivJd8ZcEppb4bVQ33DlCNw0QoE7jhBwNthyE8E0eJwcyIUO1l442JWvWPCqWXmuLYGwQx4zOYao2YY3WJr7Th6cCgziYzt/a9LU1DFvX2rE+fRMLhzsAISZC2yYLGO1U257jdrsz+chqFjSLJv502t535IU5YouLHkbysfvTeae2a0qO5ToqtNcJsAYdpxF5CaBoeQpTxFKzjAfBu89r5Ipkz8yKZBp1MCdhwcVCF4diILSy+J3sKHlfYLUzNI1pUyfZxoUv6XqLjd5tj4gGEBIFdbors3STght1y6kAYfGKfUoyk4iyZuLBI8R2qbPKMwIrqmpC7EVOYwZed4Y2NU/qRiYciDOGuw99iMlJjPI1ea/gM+FXHHhq0u10z+SBpIk7idgtdmMGpQOdix+SpMQkwfoHJGxb3EE4UTHPyqq4qcCPRYK2FTmSms4uFQNMbp1F8NwVKwCXrH+KJ9I+1OC5ZvF/QhgGllPLU2h1KAS7mmxeQFzFHh9aGf5igk38kM0btusoaCSIRjQcbunFKVkGKfac+QX7CbrFF2DlYLxGOJwuAS5MGpayximk99V27Q6hnuoODCmgnAaox6Rj5X6VpWst0xg1dklxxTaqTQXA4HpHQVl0W6QxRhgbHSt1ZR0ihnaM+3ECqc0qSTSJGIOdEoETXHLMnrL9e5llZJL+KDvryoQCuaOURFGWi6NkshzUxSMoO9pcbFcF1tANlWKCMO9Y7CvBA6PivKa4SdBYcGQHAKgjK/CNjkA76y0TwK0TBtWu5JPGxA4i5gsFZe5UIJQnd0QYo7DYrFJKTHCa1AyjUj4y8dSmfDByA9AEQYHZDN2L3wyUADZKL5NLIdMEZH0UshLkAoZWcCYBANqZVGuXShIyEGHuQy105SVV6BYu2nrLiebn8sKaamR2WkY+6Miye2ZFP9vwmVv5oe9WAfyA8P6pMeOswKlbk+vMVvN03kM0d8AbZCwMfxutxqJOlDCyYaUUXitzpCc5Wk+Q+5sCxQBDDhHV9W5FlhK+7FyVssRpWIkL14ImS3peFU1T7OgjUFuyHO/SupazSC9Lwkm5xcE9utVSSTVCbkOxpYy+lQGBhkdN0UQoL+tsHyNtMniRfQTfANshVoJPDF1NzBAnaobotZrAbD2lfUDb+7lItAuYFED+GGSptEybVO2mZfQUc7/jotMLKAJAaVprouWA4mpDhzEADgDhQ7NiKGbkPEMllkg5LP1EPEAseOXJc7dTyjn4DnvelCB5a/jnCBssLbzsyeEHf6mSI6o1w6vv0AE+JHodRr4648QMCCGPmgnNvJKjKSvJt7McY5gHMy1QTIPG/i6iAwW9SVzLiYWWg8j64JQx9dm3SkvMXkmfptIB8U1BLkhu2FfOJ6Ki0hIH2BxN4j1DxEB6lGiMAgfX+BiPIpoSbElt5gRV3hGNMM8iSQ7/YiPRTssV7A0phZfMlTjWBUDUFEG5NOdc6NSCXjHQk9c2MeY4Pc1wch+L7oDtnBtQcxUtp0V5x8LLuWnX4RGLW2/alRwCEezLn+IOskCKCNm4MsqTrJeBdZFNucwLS/r1AkkKXyqOO8iyJdF6q2a/AYtc94NTSmt19aqdr1IZNZqZpS60zFf0O2qGfR9aMcW5belqTdvKHtWGpSePBVhFGVmerzVnMnBBZSpVskLqQqk0GCzAq6hEVtzDiT590y7YhFNzPOofi3UcnMJLM+i33eSG9I47lszta7AzgINFoIuCA1lzqaQn6471uY2R3UXL0KUFs7DYsnQonD7Vq1OkkpuzK+I9Ln6fW/xnu931cHzPpQWCQ9oc4cURuI0AbBl43PG1sUYRiqGEuAhD4ZOsvrDWiEwY+jw3XQ4guJ+S+poZf9JP69Mke+SSk4EzwnMLrcyOhcMNoY1m+3cKwF+6omWNfpI7FNCscvihUoL1CD7Jgeh/+icpbHKISDNB82gvAftNeaxNuMu+JkZWeaHCcUpU4FILpRxuiiIGj2oHVqvhHCdN5ojE0KzAA0tV5fHIc595jts+XY35eeY54yvf8zxcRhfiWhhZSMbmfkYcNfaETINnLvwF1O1mE/BLk9E1Qb6Hf91uqh0qjcBl0VrZnFdQUiVfB8zxl9V65LR5Bg9j7Ap5CNZs7Qo/zGAq0q9B+Oo7V8xCEEYccB93JTxbPi+DEDftwNOakIcLpLTTg095A4Okf11kwK/HNxUEmo9lmVDL9qM+jaXvloHL6AFP4DnO86uruIrupH9kkE3I3pvNrI/Rh7fi91sxmKEbeW9387NnA90s9xUMygdTEPQpgJbBHIgRZ8W24AgkuXK+NB8QyulzAXT4QSQELlBxLwmoFaVPXkoZiPoOSYnZfmH8gI5g691l5NBAxVWCpYgstzioveLw0+dq+bBUG8Lk1heUscCT3gvuDuNyFjQMSdCCAN0hKfisNNbY5MaEjkj6lki2CrzUXDI511aE2HIA+SJdOuu0WmfJyEeOlVnRMMdmwAZc1NbzfE80W9xmR05jbO4K1WXC2sahDC94+AwZMLptNREs0R53i8ACT73LQDdfhcx197cwcQXPbIEAmIyb+p2SqmEOtdUiJX80/O+NbIFpCnVNnC6f8e4OzESDoRu/g4ClNjJDBK2cURXlF9UOwl2OrVlqiE8kHzV5plyDiKcUygY3itRFtpfWXQhuVUxFtAePAaz+WrltWbpLScin4ss/vX75X2/ftRV8LExW+xsX42QcfFkJRGkvM4C4OeQUzhZTAJi+Lrrfx1cXxlpvGQ4zHCNIjOw0lyhYSjG6qam/hSQsJTRy7amUTvDVVQVz3W0i9zfEcSajFVpszDFVCOYbBRkmoqjS82iBibc73P4RxTTPRop6GU5JXLzBAAQTQj+BOATtfaB09ipLdrV2xaXb8R1T/K5KIf5JXsl29YK2vReOElYi3bSr4eA4UA5dptE2GJSmtJtRdwGGJN10wlRPI/hNWqtGrTV2e9H8bWfR3Pw88c5/5trtYunPP1+cQXP5029CRZ1j2sw+qOAFJ7VAd1fBnA1sBm+rNvy60b3MlqJRZvv6z2kuvflrtB4hSWsYvPDnAQM1ya4cHhwmgEfo6goVodVFnRAghaagW6KwtjYAiJofEPX34WI6QNR8oGzpcPMmk753DbIcCPKurkAmMghRxTPLtLTtQ+cC3WdrxtGkoWc8p0/PBj6RPVOLmep/Rui/IPRzSWi6BJscht4L3fpi0X4C892++Mvlck6ai7k2O/+g5f6UfPd9ASKDJgh68NVDsISZE3pnt4QTPKCn4Jf26dKeCDTCWbveCf9wxMQogsgGsTlkCUcreI0csVrCJKGX48hamw2gNBh7+kzeViUyNUajS7PEowoTccy5J0+4HHi4WHA5dIofiGVWPyxbN8CvQcz+Cw+wADS9I2CS416MWE/NBgrVKcA1bMRLst2YxFam25C/Skcb/nwGKkTJrpLFxEHZ1clvGlDwP1obf3VVN5gIXOE+FODxLs1xoO62Mi+scsm01EZZMem3iJmsD6Ew9ukmS2rRLrgEAgID6upTgosCCNxqsBnXEJW9ROgR5VApB4rsk5Ykmg3ASRD4MdNwTwfgJAj8eLrUkxbRh12JgXNRvoYALmnucEFLkfqECe0Uz54w5XEwC6noS4ynu/TGs6fWV5tKo3XJSNzBXFQ8bq8B8wjQj4GsURRcBQ4/zeDJuRpVwJ+2YFJByy2kbOA1ND8CGlo4fnqqUKg6TxlFYGB0tPW/DmHYCdgFmsb8AEjxAeLUK4x6KZrALYUY8HBnZBfz4o5XCY1uYthMi4ZKqJLYFcwLw01t2E/t+6gI9GWIlF1pcoEOPYLScbBwcWKtnCl/GpBKFCAmJ1FYbVsojLEj6W4TdxIgymH8QLU6UoqtGBWksmB7gawGY2ZUm91X7alFjsU+ZhaQsZA2bApKMULCSA5uvnSlSZsSD9pytAo29/pKwciB2A7y2dNh5Lr8XuRKtWSfBsYfTC7y6M9pTFVG0EcXSEEm7WSZRjoUAEnWRDnMYS2zEe09zO7TghFkhwOA1BmTgHVY0H5Qcxt20sAJwTCunVMuHu2s2byE1hojjVbVLlVWRcNppkd6kjGnmBxNHM8xcj7BGXgUJ5sIwm+mBp2a6ykp9vzi0ULyFcN7e3e8FTLjTEpxVoVSLp0Pmjh9NXH6HzVxfuBk9wjnOkXD+XkNRvfeCUfJgClI6Nb+iwzbI2XXKjZq+c3DrRqbDrZpFRu0U9WHLFowaNGMVZoy4HhXrvQqXGVRN7zR9UCLe4q1WfIaapJf6L8r6P/AtGmSz2qutioEsoJlOwOrwhlb5weojL6hjIPEtBbKN9TzHOi9Rst/Epjm9hQJunv3k9AFfbBR/gC76QcuEPzhdvNmyHD6w4YzOGs4LUZZltPii2E6H2wyA0FWFswYd7NnxjqnjMCQ8fmgp3z+xecckk+2jRIcpn3rmTfX+0D7xoEpNvjr4Psoe0kE/L8zlq0X+JHW8lF9k7dHR0qsqqZPlQ/kfJVasb/DbM96ffcCmYyVOfb2XuABMMewv1g+BoDxvw3x/5khfvRvO/yvt8PuB9lhgZntazr8QUfYlC2dkgEihaEMF65K5vUeBFzn9HDv5C1tgCw4xrxkuy7X/qF1F3Fb20jT+nOvqILPPCTaCdGEhZ+jstITlON+YWuHLQwVDTNW8N0q2RDxvguvgX6FJ7W7HSvUcg9Lgwvq7ZZBGC2yfeIGVDKnd5UBwfUvueqAQ4tncZSZ1Na7gM5WaZz8NbR64Tsqo6eJuzzMZZEFGfQggxbSY93d+aE0nL7U5rUftoiP050/NtpxUad2gaoTqDpB2DYBdYKx0aKrhr3cN38Nda9e6Kf5aB1M175zNdr5UNcxK/yt13lALnFN1sEcyNNvvpznZYErMbiVNduvkmj3ahul9h41llg/9ORTIBeO5BFYl/JfpF5uksf04ISCF5VwB0B7EFYuMbW7NK1FJhp5WpGTBaCM1R5iMFrspO1TMDtGuJOhppmi0Rkp2muQ0z5+PDRR65Pfez76Tosh1on3drnM3EBq0oOsll0XT+RTEIp30thwQc1TNkoxb2IDDvKJAFJaENv6UnyHm2D3dD6ND6jyMhQdI1AtyH1/mN2s21MZeaEC0TLCxTR58KMlnoagPSpLu7flnjPagt0aCGRHWRVlUkGrSIa1LKA4juN1G2X7RJ2FTW/yoqK9z+0wAzvUQIeCS4AHtOJgnC9GVridTROtH8PeGe4CofopLuRlvCtWztgJ7ynJkpvaRIlORrSiFcUErZxeHeTtu1RP9qal+EnAZwo7TXDb6hYN3E2GI2uIY/LjPsrE516c3CDBeICbNwDHF8Zq4jd0tJHXR3HTIQ8WdEnCurjrZbPPaPzWxW7FKWLdI95rTrvrLQGEMcNNgrQBRZ59R0zTdVS3XZ89rOuKZa2jxH4QdRMBvH4H9T5lFi5UWexphKFAErfHkNKq5pNKxMLO6g78fUmHrVFTYyli7TkcyorrowS46SKNcSdmJMAS8KnsVF5I4Yo3n4BoA8BR4MJ7Za/ZSg2lgzGhUlf7nIwGCi1gWmpQto3ApNvrm1c1+O1dDd2D1LpCYFYIzlVQRhWXGhR+i8SV8UnZ/bdVOeudDp/O9MfBk+Om5iqX/MkTo0zySE5+2lUyz5Z7vMdVvYL7NPncU9OlXLM7g7MD6n1oWy1ihciYc6hE2yPdQV1ik3KmYWDg1Lcbx6LA6qzR3bMt9Cp4v4wC1djcwqJKmQl4Uj/3jZ1zKC9LR8sa0GBdGDDxl/PuNSL2jQJuF75TECyXyj0gb4A9A6U9al2e6ApO0RX8S+kKVn3CcBv7BVF2XbIG2vJ3+ZNUnOsy6H3mMdIgSvq+9SOeCcOHSrVrVA1b39rA5epx5pGljZ7WH0MNOo0J2T1H9pTQrs6SeFr6DFJXXVJXklRJIlogfsTdCK4hogNNIYNNAoOP52HQ5WEwyMPgl/Iw+OU8DLo8DCweBh/PQ5DVP+Vx1Ia7LEwu80PlMAho1QFaMdBKr2neXKNcLijcZbTOuKsmY9+bcBN40YDiJ4bBD6hxVDW8S29ptBp0W52eQhIYzVqtnquhmu20CoJttLqyKVc6r6lfyYbbVu+vQQ13+9ptdXoOSSCbtVq9rwY227bKYrUKR3JS+vlnOZ05T550mfYHzy5TyP9gTGUywbCQ8rJ0Fzy6S5dbku8r+CDZrK8bmD+Mho9o68EtnOvRw/vRbeTBuDQGXUECGLPmv2ioAsW+QJEVyHfdxeDXGqpzbf1qQ/WwfjxsqM7RaQ8VZ6MdTMjFfMVUM3IuOrEIhBdS7eayFTm0bpXc0hkT+dmRV1dad6y196uFuNRBB1PUKWls7i0+h/q+Nc60G64Wlqq71+hlmi6tpI++GQi3eOloyti6+ggmINN54/vnJCNGQ9soD9OBTZiXB2c8CH2cDEEfmeW0DYDcLIqs6v1ulDhWHvPMnn+VqJKL71djYSekONsU8TI/p2PphKVaQ9jkfCsOXi1ZywyOXAFSJ5fl4oTMIQAeDMxpKcpKd9R0v4Te01dT002bRgAirSMXuF5Rp3jzkIbwXfO4hAbQ3wNXIRXrrFj/gBuG9ceZ8RFveUsqCwgi6Xbo24sKlWgskJfEVcqbP2SDBSLqQiLZnHKJk1Kmd+S9dEglr8v5ZcBRVW89jYe0VHEyYGPATmypMdhBp99+mM4+7Qevba/DqX8KiQ0nzuKcnYQ9Bzc9gfM+blTmJjo6bdIfKbnp7BFymVmnfUCJS6hFEbRlWI3zxqZB2eRkDFFaCPSNUpBwpEbkCvRtnLZXhRECdHNksxO7InhOE2gffBt0bM7DHG3bpQ9hnNhz3V71Cj55wfemPpJ9sQ7/XTx6ZOyGNjxorrtIgboUWneMdWAre0hpLRoCQCXE1hiOLgrRotgey6JJ8n2dXBk7Ah1AsdiOyTGcANn6ktntmFw/o+wIKH9aXl2to2y9R0a057A/qelCDb5pDbc8GacraXdSEI++uQbL8OY69OmADs9aUEinUnzPk8d2ZPn0zfW0901/fdP/AlK7TbO4gnlEzmzq8AyfUnqv6am3SZYBPSqXoDIDtMbwzTX/D3Qi7kf2TdGPHqnOECgDdonDNbUObmeITA9P1Oj93T5edYSf6OCLvFx339BKhDrUjYR3umZR9Oaau6Bbs6jHJfW9XMhOc0K83jd416wiTXUgzc91oNuFelvhcgiibTuBVwfglZtGY2YvhMb2Xj+lm9FvFeYvfKtBvWFaA6BbYEIYRKVEFJ3ml8RgurhQ9+/W+u6nKI7FDnO7KfC/KYboNFp43yMZuQ//TNlK5cnVv/w4qtOfIIZ2eTE2VIJKNkKWwTRB6jpH0Kuge5U4Puk+1j/uoyphpHixoZZxHKAQNfXpM7BbqlQybIoXTbvfyweDYcaBMsJ5ebgKpogKbKnLJUdd8n17kqynWm3X5Qlk4+ixPmcMKlUF5BqFM2ZELrPdle88z2UiuwoIeyWzZPClkhmxoHv2D4/+VbpDfLjlH5IA2ey08qdQUZ8F7BgHdXTRmGpakeq0lMIcM64M/nUarPAqUrq53eCw1KXc15ePQo+m9N9UInScKXyu/GHd0qw2Ba792aFRdTqYMPLgNLXBGWqDltqAqJXYkNRgbPDz4aTaJwR3aVUV1cDkSTMmrek2OElF8ppXPOEZ4kE4qmeeGVQrZ3TVz99cXjxUTnNnmchYZeMlTnUpEF0pLOMcPv2KC024g558cnUTOV0iKo82XhjLQFF4r1PAE/IUZ9H2YNWsDXEmdpchuuUbnCC4mnY++cv2+sRIHh6aRnxwyI34BIm8m9A4wzVIIRHhal/FIA13EHAcSDAOh0QUFMmA0nJ5sIYVIQ2dSyspIrKvu6/4jB/74b06Ae7+S+kezvPIA4pEUW5AetL4gKaGzi/Ls6ZQ9EUIXyf+Cy5HH/ccGkQywQ0nGTGdryzVPSEULt8tHh+WdnDYus1/xIlkFLlr95UTrsm5eoXqJG8hdibwc80/6ROp2trGBpbvJVBxPSoLt+SYUstdswXLiHs/ygLG/gp+eirpCzZTtcPFE34CXisI89AOU1oF0AQdENsG0xKF0FG+OLnSVDKuZuw067cSjxt37mkfYhv2ILV7UHphmS5UprD08c1XbwG+BertlkBnyxb61pclukajYHStRsHMVEmugrcRogwfHx8/eYKowseHx87PP8vSSpdGRulxsPRwD2w1CAulxllAjHQ9SiGwdgGMCzVc6AD885zl3ID1XJluULBQcEtwCK+WwHJgRq65BVMtjy2/BSGNmny7JVCDt/mtL0t0jUbB6FqNgpmZhkbSSU1DN7PHZi8L2txIdLDUyG9zKuuBE/kktR1wpLi7BKrADewQWq5qvL0e/zti/kSeq+BrNAmB3HjQooVRbJHmcqzMlVRkhQ2l2IGlP/9MTwPrs1KNcs8Z53RrkXrMA0lpxGsy+GjEZurPvGMLjDqocf0qc6apOkMTD8nEUG/JP2T6xNBzm63YnCS2bVmYTQ+OhlFtoI/zXKrhr8Rs3bY8T8c9sUTDIOZg81oyhnGMjBfnLC5iRwfV/aOgx8Fk/knCO8w/iC7lBpIRvUx5DrgHWZcL81/K+yu7vY5aGq0eOz0YGRxwLCxt+QneHz9K/i02UYOfe9MzbXaG4NgR/Q7ObifO6cTRtkVVzxb9yupBMZfUce0x0OuUn8hjMHmttv3dpzT3YDY0agh/T5EePpi2BfuQ7rW2bEC1bMV6AGIpTsPobe4Nqtr9o7h4cPc6+mJSYWnMfRgNvTytjcYIHs0unh8/OXoP7NIJpTRHz9DK+cORT5WXfBq5rae9kbtv3O51VEyyWxdogNEHi8f3cLjVjhMYSeI7rJvbtYbgFckYJ6AHaG34z2/R8FX73MXfJaSd/4gyOhS7YPkVADgOIsA32ksHJV/IdVdvjqVf2IWRWai3un1uwaqFW//33pThFWrjC5X/wS6OJp0PuoHgM6tcUTP7FFt4nGB24vFwMCjXUMhxh4DaDoN8TnFZ6ymlPu1f+rOBzzP9Oeh9piZU8Bahd0eSHsw4DvQ5qwN48cKBYAwfxlhoRZ647knhMEbPIsRfWMSBLz7jb7m6eIQfvghF+5u3XiiQeaf6RPi6+kS0kbPV4I+jgxMexgfjF9PdJM1/oNolueJJCN24PE5L//IItMPzAZ8Pjl3J4LYTQlwZlsELby7Fka5DgVok0gHpvI86b6PQ54AkFvBKfFKFgJyEwB1RrQlhgPflc32TSq1OBvJRG3n6RO5PNq7JskebI3YcSRobojHQsbtPdFJbME6dpnrXZAy1Z9x81IrhAIPtMgKjczn62KOP+7ZnOp1WtfdHmVdp0tWGpSvu0rjZht7ljG/H1OkfXCChjCXd8+TY1+0t0rF/JXcD+FPjxcNfurl0GDluy+ws5DOYgCCY2pU3K75X19PTCiMvS2tiGRfIZxkI16QWK+ECvITR9YTY7rNsxP0xc7Blm0yVqfKWiH7Gtgzug7aTpu3p0bL3WyPlL6yUv6uSfuHOEm/hwUo13Z62lBlTeUs5befQqDULD+7hHzAwoY//p3lIij4ajQ5iKqjEEb8TI4LRJQ5odK+oBwVA9PCcfuVShZftqt91KFdcVphXjfCcjpFJymO6ZAhGZ8U2CC9TXSzJ/gh8wTewO8QzunqzppWrCCo4rkAhQ4POb76D6jQfBlMw0ASBEaE5XkVP4ERzmVTifwHzmgxZ).
![image](https://user-images.githubusercontent.com/29681384/119919283-76e68b80-bfad-11eb-8213-3dfce6b03d8c.png)
```openscad
radiiPoints=[[10,0,10],[26,20,1.1],[8,20,10],[0,7,0.3],[5,3,0.1],[-4,0,1]];
radiiPointsNegative=[[11,10,0],[15,15,1.1],[11,15,1.1]];
negativeFilletsForHole=-1.5;
extrudeHeight=2;
difference(){
    polyRoundExtrude(radiiPoints,extrudeHeight,0.5,0.5,fn=8);
    translate([0,0,-0.005])polyRoundExtrude(radiiPointsNegative,extrudeHeight+0.01,negativeFilletsForHole,negativeFilletsForHole,fn=8);
}
```

A big assumption under the hood with `polyround` is that you'll be making a ring/closed-loop. It might be possible to take that assumption out of the code, but it's pretty old code. if I were to do a re-write I would change a great many things

Does that help?

**Reply by echo-bravo-yahoo** [reply]
Absolutely! This helps immensely, thanks - I read the docs for negative fillets numerous times without putting together how to use it like this. An example in the docs would really help make this clear.
